core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -76,6 +93,16 @@ oppia.directive('collectionNodeCreator', [
             );
           };
 
+          var convertTypeAheadToExplorationId = function(newExplorationId) {
+            if (newExplorationId.length > 1 && newExplorationId[0] === '(') {
+              var explorationId = '';
+              for (var i = 2; newExplorationId[i] !== ')'; i++) {
+                explorationId += newExplorationId[i];
+              }
+              return explorationId;
+            }
+          };
```

Comment:
In practice, do you expect a failure here (e.g. if the first "if" check fails)? If not, suggest throwing an Exception to catch that.